File: Proyecto LP/mi-servidor/Frontend/lib/Enums/Estado.dart
enum EstadoEnum {
  nuevo("Nuevo", "NUEVO"),
  usado("Usado", "USADO");

  final String displayName;
  final String backendValue;

  const EstadoEnum(this.displayName, this.backendValue);

  // Método para obtener el enum desde el displayName
  static EstadoEnum? fromDisplayName(String? displayName) {
    return EstadoEnum.values.firstWhere(
      (estado) => estado.displayName == displayName,
      orElse: () => throw Exception("Estado no válido: $displayName"),
    );
  }

  // Método para obtener el valor backend desde el displayName
  static String? getBackendValue(String? displayName) {
    try {
      return fromDisplayName(displayName)?.backendValue;
    } catch (_) {
      return null; // Retorna null si no encuentra un valor válido
    }
  }

  static String? getDisplayName(String? backendValue) {
  return EstadoEnum.values.firstWhere(
    (estado) => estado.backendValue == backendValue,
    orElse: () => throw Exception("Estado no válido: $backendValue"),
  ).displayName;
}
}

File: Proyecto LP/mi-servidor/Backend/src/main/java/com/miapp/mi_servidor/Enums/Estado.java
package com.miapp.mi_servidor.Enums;

public enum Estado {
    NUEVO("Nuevo"),
    USADO("Usado");

    private final String displayName;

    Estado(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    // Método fromValue
    public static Estado fromValue(String value) {
        for (Estado estado : Estado.values()) {
            if (estado.getDisplayName().equalsIgnoreCase(value)) {
                return estado;
            }
        }
        throw new IllegalArgumentException("Valor inválido para el enum Estado: " + value);
    }
}